app/models/concerns/profile_searching.rb

```diff
@@ -38,6 +41,19 @@ module ProfileSearching
       test_results = TestResult.select(:profile_id).distinct
       has_test_results && where(id: test_results) || where.not(id: test_results)
     }
+    scope :has_policy_test_results, lambda { |has_test_results = true|
+      # the use of default scope is to bypass Pundit
+      # and avoid the situation with bind parameters
+      # that must have a different order in the resulting query
+      # of a scoped_search
+      policy_test_results = default_scoped.joins(policy_object: :test_results)
```

Comment:
Instead of `default_scoped` with a comment explaining it, why not use `Profile`?

```suggestion
      policy_test_results = Profile.joins(policy_object: :test_results)
```

Reply:
It would still have to be `Profile.default_scope`. (Try it out with `.scoping` wrapper)

Reply:
Ok, I see now. :/ That is frustrating. The `default_scoped` thing is kind of a red flag because it makes me think we're exposing records to the user that they shouldn't be able to see, even though we're not in this case; it should be fine since we're plugging it into `where` below. Still, I've looked for ways to get rid of it without success :(
Thanks for considering my ideas regardless!

Reply:
I've tried my best to pass the parameters from the Pundit-scoped case to the scoped_search, but it has to reorder (reset) used placeholder binds. I can only think of a way to compile those placeholders right into the WHERE clause for the scoped_search.

The only think that can happen here is if a policy would have two profiles, each in different account. Then it would call the policy as having a test result regardless if it is coming from the profile belonging to current account or not. Technically that is a correct result.

Reply:
Yes, but the `where` returned has the pundit scope applied, so it should filter out any profiles that are outside of the current account.

Reply:
Exactly, and that is the problem. ;-)

Check out this example
```
Profile.where(id: "fef2c74b-ef23-4dcf-b2a7-58a7a47ccdb3").arel.where_sql.gsub(/^where /i, '')
=> "\"profiles\".\"id\" = $1
```
If the `$1` gets into final where query of the scope_search, which also has the Pundit in effect, then it would do a wrong substitution.